libs/vkd3d/device.c

```diff
@@ -2613,39 +2613,33 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateComputePipelineState(d3d12_d
             &IID_ID3D12PipelineState, riid, pipeline_state);
 }
 
+static HRESULT STDMETHODCALLTYPE d3d12_device_CreateCommandList1(d3d12_device_iface *iface,
+        UINT node_mask, D3D12_COMMAND_LIST_TYPE type, D3D12_COMMAND_LIST_FLAGS flags,
+        REFIID riid, void **command_list);
+
 static HRESULT STDMETHODCALLTYPE d3d12_device_CreateCommandList(d3d12_device_iface *iface,
         UINT node_mask, D3D12_COMMAND_LIST_TYPE type, ID3D12CommandAllocator *command_allocator,
         ID3D12PipelineState *initial_pipeline_state, REFIID riid, void **command_list)
 {
-    struct d3d12_device *device = impl_from_ID3D12Device(iface);
-    struct d3d12_command_allocator *allocator;
-    struct d3d12_command_list *object;
+    ID3D12GraphicsCommandList *object;
     HRESULT hr;
 
     TRACE("iface %p, node_mask 0x%08x, type %#x, command_allocator %p, "
             "initial_pipeline_state %p, riid %s, command_list %p.\n",
             iface, node_mask, type, command_allocator,
             initial_pipeline_state, debugstr_guid(riid), command_list);
 
-    if (!(allocator = unsafe_impl_from_ID3D12CommandAllocator(command_allocator)))
-    {
-        WARN("Command allocator is NULL.\n");
-        return E_INVALIDARG;
-    }
+    if (FAILED(hr = d3d12_device_CreateCommandList1(iface, node_mask, type,
+            D3D12_COMMAND_LIST_FLAG_NONE, &IID_ID3D12GraphicsCommandList, (void **)&object)))
+        return hr;
 
-    if (allocator->type != type)
+    if (FAILED(hr = ID3D12GraphicsCommandList_Reset(object, command_allocator, initial_pipeline_state)))
     {
-        WARN("Command list types do not match (allocator %#x, list %#x).\n",
-                allocator->type, type);
-        return E_INVALIDARG;
-    }
-
-    if (FAILED(hr = d3d12_command_list_create(device, node_mask, type, command_allocator,
-            initial_pipeline_state, &object)))
+        IUnknown_Release(object);
```

Comment:
Shouldn't this be ID3D12GraphicsCommandList_Release to avoid pointer warnings?

Reply:
Fixed.